### PR TITLE
Move Markdown docs to the Markdown stdlib

### LINKED ISSUE #27094: Markdown page is empty
There's an empty page in the v0.7 docs:

screen shot 2018-05-14 at 11 35 31

from the file https://github.com/JuliaLang/julia/blob/master/stdlib/Markdown/docs/src/index.md.

I don't know what should go there, though.

### PR DESCRIPTION
**Pull request by fredrikekre**
Move docs about Markdown from the Documentation section to the Markdown stdlib, fix #27094.

### CODE REVIEW

**Comment by iblis17** [comment]
FreeBSD CI failure is unrelated,
and this error appears in https://github.com/JuliaLang/julia/pull/29986#issuecomment-437613090, also.

**Comment by fredrikekre** [comment]
#29923 x 2

**Comment by fredrikekre** [comment]
cc @mortenpi if you wanna review.

File: stdlib/Markdown/docs/src/index.md [hunk 1]
```diff
@@ -1 +1,381 @@
...
+
+!!! tip
+    As with literals in the previous section, if literal backticks need to be written within double
+    backticks use an even number greater than two. Note that if a single literal backtick needs to
+    be included within ``\LaTeX`` markup then two enclosing backticks is sufficient.
+
+!!! note
+    The `\` character should be escaped appropriately if the text is embedded in a Julia source code,
+    for example, ``` "``\\LaTeX`` syntax in a docstring." ```, since it is interpreted as a string
+    literal.
```
Comment (mortenpi) [inline stdlib/Markdown/docs/src/index.md]: Should `raw""` be mentioned here too?

File: doc/src/manual/documentation.md [hunk 2]
```diff
@@ -18,6 +18,11 @@ use indentation and code fences to delimit code examples from text. Technically,
 be associated with any other as metadata; Markdown happens to be the default, but one can construct
 other string macros and pass them to the `@doc` macro just as well.
 
+!!! note
+    Markdown support is enabled by the `Markdown` standard library
```
Comment (mortenpi) [inline doc/src/manual/documentation.md]: Doesn't "enable" here slightly imply that you should do `using Markdown` or something? Could say "Markdown support is implemented in `Markdown`".

File: stdlib/Markdown/docs/src/index.md (hunk 1 above)
Comment (fredrikekre) [reply stdlib/Markdown/docs/src/index.md]: This is just copy pasted from the other file, but I will add that.

**Pull request merged**